### ISSUE TITLE
ScrollView does not render some content

### ISSUE THREAD

**Issue by mtcairneyleeming** [migueldeicaza/gui.cs#163]
I have noticed that if I place several FrameViews in a ScrollView and interact with content inside one of the FrameViews, the content itself is redrawn, but anything else within the ScrollView is cleared and not redrawn. 

I found that if I commented out the `Clear();` call in the ScrollView's `Redraw(Rect region)`, it no longer removed any content, but instead it left artifacts after scrolling.

It can be reproduced in the example below by pressing Tab, and I have found this behaviour on both the latest master and the latest NuGet package.

```
using Terminal.Gui;

namespace TestProject
{
    internal static class Program
    {
        private static FrameView GenFrame(int yCoord)
        {
            var optionsArr = new[] {"One", "Two", "Three"};
            var frame = new FrameView("A generated frame")
            {
                X = 1,
                Y = yCoord,
                Width = 60,
                Height = optionsArr.Length + 6
            };
            var categoryRadio = new RadioGroup(1, 1, optionsArr);
            frame.Add(categoryRadio);
            frame.Add(new Button("Check")
            {
                Y = optionsArr.Length + 2,
                X = 1,
                Width = 7,
                Height = 1
            });
            return frame;
        }

        public static void Main()
        {
            Application.Init();
            var top = Application.Top;

            var window = new Window("Test")
            {
                X = 0, Y = 0, Width = Dim.Fill(), Height = Dim.Fill()
            };
            var scrollView = new ScrollView(new Rect(0, 0, top.Frame.Width - 3, top.Frame.Height - 2))
            {
                ShowVerticalScrollIndicator = true,
                ShowHorizontalScrollIndicator = false,
                X = 1,
                Y = 1,
                Width = Dim.Fill(),
                Height = Dim.Fill(),
                ContentSize = new Size(160, 400),
                ContentOffset = new Point(0, 0),
            };
            window.Add(scrollView);
            scrollView.Add(new Button("X")
            {
                Clicked = Application.RequestStop,
                X = Pos.Right(scrollView) - 8,
                Y = 1,
                Width = 5,
                Height = 1
            });

            var generatedQuestionFrame = GenFrame(1);
            scrollView.Add(generatedQuestionFrame);
            var generatedQuestionFrame2 = GenFrame(10);
            scrollView.Add(generatedQuestionFrame2);

            scrollView.Add(new Button("Finish")
            {
                X = 1,
                Y = 30,
                Width = 8,
                Height = 1,
            });
            Application.Run(window);
        }
    }
}
```
P.S. I love this framework

**Comment by tznind** [comment]
Any workaround for this? I find that even with 2 labels and 2 text boxes I get this issue (on windows) 

Upper image is after click in top textbox and lower image is after a keypress (other controls vanish)

![image](https://user-images.githubusercontent.com/31306100/69191255-0ccc6b00-0b1a-11ea-9f04-c93f866c454a.png)


```csharp
Application.Init ();
var top = Application.Top;

var scroll = new ScrollView(new Rect(5, 5, 20, 20));
scroll.Add(new Label(0,0,"Name:"));
scroll.Add(new Label(0,1,"Addr:"));
scroll.Add(new TextField(5,0,5,"-"));
scroll.Add(new TextField(5,1,5,"-"));
top.Add(scroll);
top.SetFocus(scroll);

Application.Run();
```

**Comment by tig** [comment]
Fixed in #586
